types/three/three-core.d.ts

```diff
@@ -6440,17 +6551,37 @@ export class EdgesGeometry extends BufferGeometry {
     constructor(geometry: BufferGeometry, thresholdAngle: number);
 }
 
+export interface ExtrudeGeometryParameters {
+    curveSegments?: number;
+    steps?: number;
+    amount?: number;
+    bevelEnabled?: boolean;
+    bevelThickness?: number;
+    bevelSize?: number;
+    bevelSegments?: number;
+    extrudePath?: CurvePath<Vector>;
+    frames?: any;
+    UVGenerator?: any;
+}
+
+export class ExtrudeBufferGeometry extends BufferGeometry {
+    constructor(shape?: Shape, options?: ExtrudeGeometryParameters);
+    constructor(shapes?: Shape[], options?: ExtrudeGeometryParameters);
+
+    addShapeList(shapes: Shape[], options?: ExtrudeGeometryParameters): void;
+    addShape(shape: Shape, options?: ExtrudeGeometryParameters): void;
+}
+
 export class ExtrudeGeometry extends Geometry {
-    constructor(shape?: Shape, options?: any);
-    constructor(shapes?: Shape[], options?: any);
+    constructor(shape?: Shape, options?: ExtrudeGeometryParameters);
+    constructor(shapes?: Shape[], options?: ExtrudeGeometryParameters);
 
-    static WorldUVGenerator: {
-        generateTopUV(geometry: Geometry, indexA: number, indexB: number, indexC: number): Vector2[];
-        generateSideWallUV(geometry: Geometry, indexA: number, indexB: number, indexC: number, indexD: number): Vector2[];
-    };
+    addShapeList(shapes: Shape[], options?: ExtrudeGeometryParameters): void;
```

Comment:
agreed, don't think it should be deleted, and in fact you should extract its interface and share it with the `UVGenerator?: any` above.